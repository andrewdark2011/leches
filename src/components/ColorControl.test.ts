import { mount } from '@vue/test-utils'
import { TresLeches, useControls } from '/@/'
import { it, expect } from 'vitest'
import { defineComponent, nextTick } from 'vue'


describe('Color Control', async () => {
  it('should render a color control', async ()=> {
    const component = defineComponent({
      template: `
        <TresLeches />
      `,
      components: {
        TresLeches
      },
      setup() {
        const { value: colorValue } = useControls({ colorValue: '#008080' })
        return {
          colorValue
        }
      }
    })

    const wrapper = mount(component, {
      components: {
        TresLeches
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
    expect(wrapper.find('input').attributes('type')).toBe('color')
  })
  it('should render a color control with a label', async ()=> {
    const component = defineComponent({
      template: `
        <TresLeches />
      `,
      components: {
        TresLeches
      },
      setup() {
        const { colorValue } = useControls({ colorValue: '#008080' })
        return {
          colorValue
        }
      }
    })

    const wrapper = mount(component, {
      components: {
        TresLeches
      },
    })
    expect(wrapper.find('label').text()).toBe('colorValue')
  })
  it('should change the value of the control when the input changes', async ()=> {
    const component = defineComponent({
      template: `
        <TresLeches />
      `,
      setup() {
        const { value: colorValue } = useControls({ colorValue: '#008080' })
        return {
          colorValue
        }
      }
    })

    const wrapper = mount(component, {
      components: {
        TresLeches
      },
    })
    const input = wrapper.find('input[type="color"]')

    await input.setValue('#ff2400')
    await nextTick();

    expect(wrapper.vm.colorValue).toBe('#ff2400')
  })
  it('should hide the control when the visible property is toggled', async ()=> {
    const component = defineComponent({
      template: `
        <TresLeches />
      `,
      setup() {
        const { value: colorValue, visible } = useControls({ colorValue: '#008080' })
        return {
          colorValue,
          visible
        }
      }
    })

    const wrapper = mount(component, {
      components: {
        TresLeches
      },
    })
    wrapper.vm.visible = false
    await nextTick();
    expect(wrapper.find('input[type="color"]').exists()).toBe(false)
  })
  it('should show the control by default', async ()=> {
    const component = defineComponent({
      template: `
        <TresLeches />
      `,
      setup() {
        const { visible } = useControls({ test: '#008080' })
        return {
          visible
        }
      }
    })

    const wrapper = mount(component, {
      components: {
        TresLeches
      },
    })
    const checkboxInput = wrapper.find('input[type="color"]')
    expect(checkboxInput.exists()).toBe(true)

    expect(wrapper.vm.visible).toBe(true)
  })
  it('should not show the control if the visibility is initially false', async ()=> {
    const component = defineComponent({
      template: `
        <TresLeches />
      `,
      setup() {
        const { visible } = useControls({ test: {
          value: '#008080',
          visible: false
        } })
        return {
          visible
        }
      }
    })

    const wrapper = mount(component, {
      components: {
        TresLeches
      },
    })
    const checkboxInput = wrapper.find('input[type="color"]')
    expect(checkboxInput.exists()).toBe(false)

    expect(wrapper.vm.visible).toBe(false)
  })
})