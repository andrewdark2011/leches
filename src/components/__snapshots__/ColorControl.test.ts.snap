// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Color Control > should render a color control 1`] = `
"<div style=\\"left: 704px; top: 10px;\\" class=\\"absolute select-none z-24 w-280px font-sans text-xs\\">
  <div tabindex=\\"0\\" class=\\"bg-white shadow-xl rounded border-4 border-solid border-black\\">
    <header class=\\"relative cursor-grabbing p-4 flex justify-between text-gray-200 relative\\"><i class=\\"h-4 w-4 p-1.5 flex items-center line-height-0 rounded-full bg-gray-100 text-xs\\">🍰</i>
      <div><i class=\\"i-ic-baseline-drag-indicator\\"></i><i class=\\"i-ic-baseline-drag-indicator\\"></i><i class=\\"i-ic-baseline-drag-indicator\\"></i></div>
      <div></div>
    </header>
    <!--v-if-->
    <div class=\\"flex px-4 justify-between gap-4 items-center mb-2\\"><label class=\\"text-gray-500 w-1/3\\">colorValue</label><input type=\\"color\\"><input class=\\"p-2 w-2/3 rounded text-right text-xs text-gray-400 bg-gray-100 focus:border-gray-200 outline-none border-none font-sans\\" type=\\"text\\"></div>
  </div>
</div>"
`;
